src/serializer/ResidualHeapVisitor.js

```diff
@@ -1306,8 +1306,14 @@ export class ResidualHeapVisitor {
     this._visitUntilFixpoint();
 
     let referentializer = this.referentializer;
-    if (referentializer !== undefined)
+    if (referentializer !== undefined) {
+      referentializer.helperState = {
```

Comment:
I really don't like this injection from the distance via state mutation. The challenge is to find a way to refactor the code to pass this information in from the outside...
Also, the `referentializer` is only used by the loop below; not clear whether that should be part of `visitRoots` to begin with. Move it out.